### ISSUE TITLE
When a file is not buffered enough it exits

### ISSUE THREAD

**Issue by heartlogic** [scakemyer/plugin.video.quasar#756]
## Expected Behavior
When we choose to play a file from Kodi library, for example an episode, it starts buffering and then it starts playing... If the file is not buffered enough it must pause, continue to download until it has buffered enough and continue playing...

## Current Behavior
Right now, when we choose to play a file from Kodi library, for example an episode, it starts buffering and then it starts playing... If the file is not buffered enough it stops playing and exits to Kodi episodes menu... 

## Steps to Reproduce (for bugs)
1. Go to Kodi library and choose to play an episode from a series added from Quasar
2. Play the file

## Context
It is annoying to start again and again the same file when internet speed is not enough to keep up with the needed episode download rate...  

## Your Environment
* Version used: Quasar 0.9.73
* Environment name and version (e.g. Kodi 15/16/17, SPMC, OSMC, OpenELEC): Kodi 17.1 RC
* Operating System and version (Android 4/5/6, Windows 7/8/10, macOS 10.11, Ubuntu 16.10): LibreELEC 8.0.0b in S905X Amlogic TV Box
* Link to Kodi debug log:
http://paste.ubuntu.com/24052231/

**Comment by elgatito** [comment]
I just wanted to investigate similar situation, I've met  few days ago. When starting a movie, and rewind somewhere, buffering starts, then you start watching, but in a minute or two kodi exits from the movie. I thought maybe timer is counting buffering is done or not, but not relative to current place, rather place you have started a movie.
I wanted to debug buffering timers.

**Comment by elgatito** [comment]
Sorry, I'm not familiar with Go.
Looking at this place: https://github.com/scakemyer/quasar/blob/master/bittorrent/player.go#L748
If I open the stream with big size, and quickly rewind forward somewhere, the initial buffer piece will get less priority, the one I've moved on will get bigger, right?
Will it cancel previous timer, set in that function?

**Comment by scakemyer** [comment]
Yeah that's not where that happens, but in `torrentfs.go`. I've yet to look into that one close enough to see what kind of improvements would be possible.

**Comment by elgatito** [comment]
Opened a new file from main menu (Estuary). When it buffered and opened - quickly move to half of a movie by "right arrow" jumps. After some seconds I'm out of the movie to the main menu.

http://paste.ubuntu.com/24057864/

**Comment by bloombergg** [comment]
Files are stored locally or LAN.  You are clearly not playing files, but torrents.

> when internet speed is not enough to keep up with the needed episode download rate..

These are two terms describing the same thing.

Did you perhaps mean the isp throuput is lower than video bitrate?

**Comment by bloombergg** [comment]
throughput

**Comment by heartlogic** [comment]
An email notification came from a user named "Bloombergg" who commented on this an other threads but i cannot see his comments nowhere when entering Github!
